src/algorithms/bfs.jl

```diff
@@ -6,24 +6,22 @@ which is sequential and runs through every vertex in the graph. If you know
 the maximum number of active vertices, call `bfs(s, source, nv)` which is faster.
 """
 function bfs(s::Stinger, source::Int64)
-    nv = get_nv(s)
-    bfs(s, source, nv)
+    bfs(s, source, s[max_nv])
 end
 
 function bfs(s::Stinger, source::Int64, nv::Int64)
-    nv>source || throw(DimensionMismatch("Attempting to run BFS with source $source in a graph with only $nv vertices."))
+    #nv>source || throw(DimensionMismatch("Attempting to run BFS with source $source in a graph with only $nv vertices."))
```

Comment:
You took out this exception throw because of type stability, could you use a return value to signal an error? Like perhaps an empty `parents` array, or  `return -1, []` and then in success `return 0, parents`? That changes the API so I like the idea of returning just an empty parents array. which is O(1) to check for success.